apps/communications/contacts/views/form/js/form_controller.js

```diff
@@ -168,16 +185,29 @@
     };
   }
 
+  function listenContactChanges(events) {
+    return new Promise(function(resolve, reject) {
+      ContactsService.addListener('contactchange',
+        function oncontactchange(event) {
+          var eventToSave = {
+            contactID: event.contactID,
+            reason: event.reason
+          };
+          
+          if (!events) {
+            events = [];
+          }
+          events.unshift(eventToSave);
+          sessionStorage.setItem('oncontactchange', JSON.stringify(events));
+          ContactsService.removeListener('contactchange', oncontactchange);
```

Comment:
Here is ok, but we usually remove the listeners on the first line of that listener, that will be in this case moving this line to L192